validator/src/admin_rpc_service.rs

```diff
@@ -324,6 +338,60 @@ impl AdminRpc for AdminRpcImpl {
     fn contact_info(&self, meta: Self::Metadata) -> Result<AdminRpcContactInfo> {
         meta.with_post_init(|post_init| Ok(post_init.cluster_info.my_contact_info().into()))
     }
+
+    fn get_secondary_index_key_size(
+        &self,
+        meta: Self::Metadata,
+        pubkey_str: String,
+    ) -> Result<HashMap<RpcAccountIndex, usize>> {
+        AdminRpcImpl::supports_full_api(&meta)?;
```

Comment:
This is the main question mark. What do you think about this approach to segment `full_api` methods?
I originally added this method in a new module/AdminRpcImpl struct (like the multiple modules in `solana-rpc`). This works great for the IPC server in `admin_rpc_service::run()`, which can be extended by both structs. But the derived clients are separate, and only one explicit type can be constructed in `admin_rpc_service::connection()`.